### PR TITLE
Dynamic y axis range for precipitation graph

### PR DESCRIPTION
**Pull request by Kyubinhan**
* Move setAxiosRequestInterceptors to AuthWrapper
* Set the max of y range based on given precip data
* Fix a bug in getTickValues util function

### CODE REVIEW

File: web/src/features/stations/components/WxStationDropdown.tsx [hunk 1]
```diff
@@ -37,18 +42,23 @@ const WxStationDropdown = (props: Props) => {
...
-  let isThereInvalidCode = false
+
+  let isThereUnknownCode = false
   const maxNumOfSelect = props.maxNumOfSelect || 3
-  const autocompleteValue = props.stationCodes.map(code => {
-    const station = stationsByCode[code]
-    if (station) {
-      return station
+  const autocompleteValue: Option[] = props.stationCodes.map(code => {
+    const s = stationsByCode[code]
```
Comment (Sybrand) [inline web/src/features/stations/components/WxStationDropdown.tsx:49]: Why rename from station to s? I would think station is better?

File: web/src/features/fireWeather/components/graphs/PrecipGraph.tsx [hunk 2]
```diff
@@ -86,46 +86,59 @@ const PrecipGraph: React.FunctionComponent<Props> = ({
 
   // useMemo will only recompute the memoized value when one of the dependencies has changed.
   // This optimization helps to avoid expensive calculations on every render.
-  const { xDomain, xTickValues, observedPrecips, forecastPrecips } = useMemo(() => {
+  const calculated = useMemo(() => {
```
Comment (Sybrand) [inline web/src/features/fireWeather/components/graphs/PrecipGraph.tsx:89]: Maybe a more descriptive name?

**Review by Sybrand — APPROVED** [review]
Minor comments re. naming.

File: web/src/features/stations/components/WxStationDropdown.tsx (hunk 1 above)
Comment (Kyubinhan) [reply web/src/features/stations/components/WxStationDropdown.tsx:49]: Okay will update it

File: web/src/features/fireWeather/components/graphs/PrecipGraph.tsx (hunk 2 above)
Comment (Kyubinhan) [reply web/src/features/fireWeather/components/graphs/PrecipGraph.tsx:89]: Oh... I wish we never come to this. Any suggestion? `calculatedValues` or `memoizedCalc`?

**Review by andrea-williams — APPROVED** [review]

File: web/src/features/fireWeather/components/graphs/PrecipGraph.tsx (hunk 2 above)
Comment (Sybrand) [reply web/src/features/fireWeather/components/graphs/PrecipGraph.tsx:89]: Soooo.... it's this stuff: 
```
return {
      xDomain,
      xTickValues: d3Utils.getTickValues(xDomain, utcOffset, false),
      maxPrecip,
      observedPrecips,
      forecastPrecips
    }
```
that's.... umm... graphCalculations ?

**Pull request merged**